mitodl/ocw-hugo-themes
# fix sitemaps

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/673

#### What's this PR do?
First of all, a few prerequisites:
 - Swaps out `hugo-bin` for `hugo-bin-extended`, which uses the latest version of Hugo (0.98.0)
 - Upgrades the version of NodeJS used in Github Actions runners to latest LTS (16.15.0)

This PR accomplishes a few things regarding sitemap generation:
 - General sitemap template moved to the base theme so they apply to all themes
 - Base theme sitemap template now filters out content with `headless` set to true (static JSON, etc.)
 - Adds a new `home.xml` template to the `www` theme that in conjunction with the changes in https://github.com/mitodl/ocw-hugo-projects/pull/176 will facilitate the generation of a `sitemap_index.xml` file.  This file will tell Google where all of the course site sitemaps are located.  It obtains a list of published course sites using the `ocw-studio` API and generates a list of sitemaps based on the response.

#### How should this be manually tested?
 - Clone [`ocw-hugo-projects`](https://github.com/mitodl/ocw-hugo-projects) on the `cg/sitemap-index` branch
 - Clone the [`ocw-www`](https://github.mit.edu/mitocwcontent/ocw-www) content repo
 - Clone `ocw-hugo-themes` on this PR branch and make sure you set the following variables in your `.env` file:
   - `SITEMAP_DOMAIN=ocw.mit.edu`
   - `WWW_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/ocw-www/config.yaml`
   - `WWW_CONTENT_PATH=/path/to/mitocwcontent/ocw-www/`
 - Spin up `ocw-www` with `npm run start:www`
 - Visit http://localhost:3000/sitemap.xml and verify that the sitemap index looks correct according to [Google's documentation](https://developers.google.com/search/docs/advanced/sitemaps/large-sitemaps)
 - Visit some of the sitemaps referenced in the index and verify that the links work
 - Visit http://localhost:3000/sitemap-www.xml and verify that the content in `ocw-www` itself is in there, sans `headless` content like instructors
 - Verify that the links inside `sitemap-www.xml` work

FYI: It is expected that the deploy preview will fail as https://github.com/mitodl/ocw-hugo-projects/pull/176 needs to be merged to add the expected env variables to the security configuration
